pkgs/games/polymc/default.nix

```diff
@@ -23,34 +24,43 @@ in mkDerivation rec {
       --replace 'scanJavaDir("/usr/lib/jvm")' 'javas.append("${jdk}/lib/openjdk/bin/java")' \
       --replace 'scanJavaDir("/usr/lib32/jvm")' 'javas.append("${jdk8}/lib/openjdk/bin/java")'
 
+  '' + lib.optionalString (msaClientID != "") ''
     # add client ID
-    substituteInPlace notsecrets/Secrets.cpp \
-      --replace 'QString MSAClientID = "";' 'QString MSAClientID = "${msaClientID}";'
+    substituteInPlace CMakeLists.txt \
+      --replace '17b47edd-c884-4997-926d-9e7f9a6b4647' '${msaClientID}'
   '';
 
-  cmakeFlags = [ "-DMultiMC_LAYOUT=lin-system" ];
+  cmakeFlags = [ "-DLauncher_LAYOUT=lin-system" ];
+
+  desktopItem = makeDesktopItem {
```

Comment:
Use the DesktopItems trio instead of a custom copy.

Reply:
done